src/tools/utils/editorOperationUtils.ts

```diff
@@ -0,0 +1,114 @@
+import * as monaco from 'monaco-editor'
+import { ValidAnnotatedEditOperation } from 'vscode/vscode/vs/editor/common/model'
+import { getLockedRanges, minusRanges } from './rangeUtils'
+
+function createNewOperation (
+  oldOperation: ValidAnnotatedEditOperation,
+  newRange: monaco.Range,
+  newText: string | null,
+  index: number
+): ValidAnnotatedEditOperation {
+  const identifier = oldOperation.identifier != null
+    ? { major: oldOperation.identifier.major, minor: oldOperation.identifier.minor + index }
+    : null
+  return new ValidAnnotatedEditOperation(
+    identifier,
+    newRange,
+    newText,
+    oldOperation.forceMoveMarkers,
+    oldOperation.isAutoWhitespaceEdit,
+    oldOperation._isTracked
+  )
+}
+
+function createNewOperationsFromRanges (
+  oldOperation: ValidAnnotatedEditOperation,
+  editableRanges: monaco.Range[],
+  splitText: (string | null)[]
+): ValidAnnotatedEditOperation[] {
+  if (editableRanges.length <= 0 || splitText.length <= 0) {
+    return [oldOperation]
+  }
+
+  return splitText.map((text, index) => createNewOperation(oldOperation, editableRanges[index]!, text, index))
+}
+
+function splitOperationText (
+  model: monaco.editor.ITextModel,
+  uneditableRanges: monaco.Range[],
+  text: string | null
+): (string | null)[] {
+  if (text == null || text === '') {
+    return uneditableRanges.length > 0 ? [] : [text]
+  }
+
+  const splitText: string[] = []
+  const uneditableRangesText = uneditableRanges.map(range => model.getValueInRange(range))
+  let currentRange: number = 0
+  let textToSplit: string = text
+  while (textToSplit !== '' && currentRange < uneditableRangesText.length) {
```

Comment:
nit: `textToSplit.length > 0` seems clearer to me (we're looping until the string is empty)

also, wouldn't something like `remainingText` be more relevant?

Reply:
[refactor: rename functions on locked code and their variables](https://github.com/CodinGame/monaco-editor-wrapper/pull/125/commits/4211c36f490bd0ebf871eeb479fac4afd597cb2d)